app/com/linkedin/drelephant/executors/ThreadPoolExecutorService.java

```diff
@@ -29,19 +29,19 @@ public void execute(AnalyticJob analyticJob) {
     @Override
     public void startService() {
 
-        long checkPoint;
+        long checkPoint = 0L;
         ElephantRunner elephantRunner = ElephantRunner.getInstance();
 
         while (elephantRunner.getRunningStatus().get() && !Thread.currentThread().isInterrupted()) {
 
-            checkPoint = elephantRunner.getAnalyticJobGenerator().fetchCheckPoint();
             elephantRunner.getAnalyticJobGenerator().fetchAndExecuteJobs(checkPoint);
+            checkPoint = elephantRunner.getAnalyticJobGenerator().getCheckPoint();
```

Comment:
This will ensure that RM starts picking up job from start in case of ThreadPoolExecutor and then persist the checkpoint in DB.